Packs/GitHubITAdmin/Integrations/GitHub_IT_Admin/GitHub_IT_Admin.py

```diff
@@ -0,0 +1,615 @@
+        self.session = requests.Session()
+        if not proxy:
+            self.session.trust_env = False
+        self.headers['Authorization'] = 'Bearer ' + self.token
+
+    def http_request(self, method, url_suffix, params=None, data=None, headers=None):
+
+        if not headers:
+            headers = self.headers
+        full_url = self.base_url + url_suffix
+        res = requests.request(
+            method,
+            full_url,
+            verify=self.verify,
+            params=params,
+            json=data,
+            headers=headers
+        )
+
+        return res
+
+    def get_user(self, input_type, user_term):
+
+        if input_type == 'id':
+            uri = f'scim/v2/organizations/{encode_string_results(self.org)}/Users/{encode_string_results(user_term)}'
+
+        else:
+            user_term = "\"" + user_term + "\""
+            uri = f'scim/v2/organizations/{encode_string_results(self.org)}/' \
+                  f'Users?filter={encode_string_results(input_type)} eq {encode_string_results(user_term)}'
+        return self.http_request(
+            method='GET',
+            url_suffix=uri,
+        )
+
+    def create_user(self, data):
+        uri = f'scim/v2/organizations/{encode_string_results(self.org)}/Users'
+        return self.http_request(
+            method='POST',
+            url_suffix=uri,
+            data=data,
+        )
+
+    def update_user(self, user_term, data):
+        uri = f'scim/v2/organizations/{encode_string_results(self.org)}/Users/{encode_string_results(user_term)}'
+        return self.http_request(
+            method='PUT',
+            url_suffix=uri,
+            data=data,
+        )
+
+    def disable_user(self, data):
+
+        uri = f'scim/v2/organizations/{encode_string_results(self.org)}/Users/{encode_string_results(data)}'
+        return self.http_request(
+            method='DELETE',
+            url_suffix=uri
+        )
+
+    # Builds a new user github_user dict with pre-defined keys and custom mapping (for user)
+    def build_github_create_user(self, args, scim):
+        extension_schema = scim.get(SCIM_EXTENSION_SCHEMA)
+
+        github_user = scim
+
+        custom_mapping = None
+        if args.get('customMapping'):
+            custom_mapping = json.loads(args.get('customMapping'))
+        elif CUSTOM_MAPPING_CREATE:
+            custom_mapping = json.loads(CUSTOM_MAPPING_CREATE)
+
+        if custom_mapping and extension_schema:
+            for key, value in custom_mapping.items():
+                # key is the attribute name in input scim. value is the attribute name of app profile
+                user_extension_data = extension_schema.get(key)
+                if user_extension_data:
+                    github_user[value] = user_extension_data
+
+        return github_user
+
+    # Builds a new user github profile dict with pre-defined keys and custom mapping (for user)
+    def build_github_update_user(self, args, github_user, scim):
```

Comment:
we can unify this method with the one above - let's go over this in the demo.

Reply:
We have made the changes.